src/php/Lang/Registry.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Lang;
+
+use Phel\Lang\Collections\Map\PersistentMapInterface;
+
+final class Registry
+{
+    /** @var array<string, array<string, mixed>> */
+    private $definitions;
+
+    /** @var array<string, array<string, mixed>> */
+    private $definitionsMetaData;
+
+    /** @var ?self */
+    private static $instance = null;
+
+    public function __construct()
+    {
+        $this->clear();
+    }
+
+    /**
+     * @return self
+     */
+    public static function getInstance()
```

Comment:
Is there any reason this class is a Singleton?
It couldn't be possible to create from the factory? :)

Reply:
There is no Factory inside the Lang module/directory. This Registry is the replacement for the `$GLOBALS` usage, so for now I think it's way better to use this Singleton instead.